### PR TITLE
Improve product browser and product ingest UI

### PR DESCRIPTION
**Pull request by pavinduLakshan**
Included changes:

1. Add progress bars to display the progress of file ingestion processes.
2. Fix product browser pagination
3. Create a new endpoint to get product types
4. Improve search bar UI
5. Move file manager REST api calls to a separate dedicated module.

### CODE REVIEW

**Review by IMS94 — APPROVED** [review]

File: react-components/oodt_opsui_sample_app/src/components/ProgressBar.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ */
+import React, { Component } from "react";
+import LinearProgress from "@material-ui/core/LinearProgress";
+import Typography from "@material-ui/core/Typography";
+import Box from "@material-ui/core/Box";
+
+class ProgressBar extends Component {
+  render() {
+    return (
+      <Box display="flex" alignItems="center" style={{ marginLeft: 20 }}>
```
[the remark this answers is not in the document]
Comment (pavinduLakshan) [reply react-components/oodt_opsui_sample_app/src/components/ProgressBar.js]: Yes, we can. I updated it.

File: webapp/fmprod/src/main/java/org/apache/oodt/cas/product/jaxrs/resources/ProductPageResource.java [hunk 2]
```diff
@@ -98,6 +100,10 @@ public ProductPageResource(
     }
   }
 
+  public void setTotalProducts(int totalProducts){
```
Comment (NGimhana) [inline webapp/fmprod/src/main/java/org/apache/oodt/cas/product/jaxrs/resources/ProductPageResource.java:103]: Minor: lets add java doc comment here and getter

File: webapp/fmprod/src/main/java/org/apache/oodt/cas/product/jaxrs/services/FileManagerJaxrsServiceV2.java [hunk 3]
```diff
@@ -137,6 +137,20 @@ public ProductTypeListResource getProductTypes() throws WebApplicationException
     }
   }
 
+  public int getTotalNumOfProducts() throws WebApplicationException {
```
Comment (NGimhana) [inline webapp/fmprod/src/main/java/org/apache/oodt/cas/product/jaxrs/services/FileManagerJaxrsServiceV2.java:140]: Minor: Shall we add a JavaDoc comment for the method?

**Review by NGimhana — APPROVED** [review]

File: webapp/fmprod/src/main/java/org/apache/oodt/cas/product/jaxrs/resources/ProductPageResource.java (hunk 2 above)
Comment (pavinduLakshan) [reply webapp/fmprod/src/main/java/org/apache/oodt/cas/product/jaxrs/resources/ProductPageResource.java:108]: added

File: webapp/fmprod/src/main/java/org/apache/oodt/cas/product/jaxrs/services/FileManagerJaxrsServiceV2.java (hunk 3 above)
Comment (pavinduLakshan) [reply webapp/fmprod/src/main/java/org/apache/oodt/cas/product/jaxrs/services/FileManagerJaxrsServiceV2.java:145]: Sure. added it.

**Pull request merged**